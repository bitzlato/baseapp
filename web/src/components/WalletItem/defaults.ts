import { ZERO_PENCE } from 'web/src/helpers/money';
import { WalletItemData } from './WalletItem';

export const DEFAULT_WALLET_ITEM: WalletItemData = {
  name: '',
  currency: '',
  balanceTotal: ZERO_PENCE,
  balanceP2P: ZERO_PENCE,
  balanceMarket: ZERO_PENCE,
  approximate: ZERO_PENCE,
  locked: ZERO_PENCE,
  hasDepositWithdraw: false,
  hasTransfer: false,
  hasGift: false,
  index: -1,
};
