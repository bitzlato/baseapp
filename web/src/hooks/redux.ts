import { shallowEqual, useSelector } from 'react-redux';
import { RootState } from '../modules';

export function useShallowSelector<T>(selector: (state: RootState) => T): T {
  return useSelector(selector, shallowEqual);
}
