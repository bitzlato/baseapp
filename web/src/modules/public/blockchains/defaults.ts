import { Blockchain } from './types';

export const DEFAULT_BLOCKCHAIN: Blockchain = {
  explorer_address: '',
  explorer_transaction: '',
  height: 1500000,
  id: 2,
  is_transaction_price_too_high: false,
  key: 'btc-testnet',
  min_confirmations: 6,
  name: 'Bitcoin Testnet',
  status: 'disabled',
};
