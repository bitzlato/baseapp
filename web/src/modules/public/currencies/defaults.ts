import { DEFAULT_CCY_PRECISION } from 'src/constants';
import { createCcy, createMoney } from 'src/helpers/money';

import { ApiCurrency, CurrencySource } from './types';

export const DEFAULT_CURRENCY = createCcy('', DEFAULT_CCY_PRECISION);

export const defaultCurrency: ApiCurrency = {
  blockchain_currencies: [],
  blockchain_ids: [],
  cc_code: '',
  deposit_enabled: true,
  deposit_fee: createMoney(0, DEFAULT_CURRENCY),
  description: null,
  homepage: null,
  icon_id: '',
  icon_url: '',
  id: '',
  min_withdraw_amount: createMoney(0, DEFAULT_CURRENCY),
  name: '',
  position: 0,
  precision: 8,
  price: '1.0',
  type: 'coin',
  withdraw_limit_24h: '0',
  withdraw_limit_72h: '0',
  withdrawal_enabled: true,
  ...DEFAULT_CURRENCY,
};

export const defaultCurrencySource: CurrencySource = {
  blockchain_currencies: [],
  blockchain_ids: [],
  cc_code: '',
  deposit_enabled: true,
  deposit_fee: '0',
  description: null,
  homepage: null,
  icon_id: '',
  icon_url: '',
  id: '',
  min_withdraw_amount: '0',
  name: '',
  position: 0,
  precision: 8,
  price: '1.0',
  type: 'coin',
  withdraw_limit_24h: '0',
  withdraw_limit_72h: '0',
  withdrawal_enabled: true,
};
