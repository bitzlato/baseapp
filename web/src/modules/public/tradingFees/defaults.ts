import { DEFAULT_CCY_PRECISION } from 'src/constants';
import { createCcy, createMoney } from 'src/helpers/money';

import { ApiCurrency } from './types';

export const DEFAULT_CURRENCY = createCcy('', DEFAULT_CCY_PRECISION);

export const defaultCurrency: ApiCurrency = {
  id: '',
  name: '',
  symbol: '',
  explorer_transaction: '',
  explorer_address: '',
  type: 'coin',
  deposit_fee: createMoney(0, DEFAULT_CURRENCY),
  min_confirmations: 6,
  min_deposit_amount: createMoney(0, DEFAULT_CURRENCY),
  withdraw_fee: createMoney(0, DEFAULT_CURRENCY),
  min_withdraw_amount: createMoney(0, DEFAULT_CURRENCY),
  deposit_enabled: true,
  withdrawal_enabled: true,
  withdrawal_disabled_reason: '',
  base_factor: 100000000,
  precision: 8,
  icon_url: '',
  icon_id: '',
  price: '1.0',
  ...DEFAULT_CURRENCY,
};
