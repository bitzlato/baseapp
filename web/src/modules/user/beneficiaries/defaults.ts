import { Beneficiary } from './types';

export const defaultBeneficiary: Beneficiary = {
  id: 0,
  currency: '',
  name: '',
  state: '',
  data: {
    address: '',
  },
  blockchain_id: -1,
  blockchain_key: '',
  uid: '',
  sent_at: '',
};
