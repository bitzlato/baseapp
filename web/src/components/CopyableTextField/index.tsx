import '@openware/cryptofont';
import cn from 'classnames';
import * as React from 'react';
import { InputGroup } from 'react-bootstrap';
import { CopyIcon } from 'src/assets/icons/CopyIcon';
import { useT } from 'src/hooks/useT';
import { CustomInput } from '../';
import { copy } from '../../helpers';
import { IconButton } from '../IconButton/IconButton';

export interface CopyableTextFieldProps {
    /**
     * Text value that will be copied to the clipboard
     */
    value: string;
    /**
     * Additional class name for styling. By default element receives `cr-button` class
     * @default empty
     */
    className?: string;
    /**
     * String value that makes copy field be unique
     */
    fieldId: string;
    /**
     * @default 'false'
     * If true, Button will be disabled.
     */
    disabled?: boolean;
    label?: string;
}

/**
 * Text field component with ability to copy inner text.
 */
export const CopyableTextField: React.FC<CopyableTextFieldProps> = ({ value, className, disabled, fieldId, label }) => {
    const t = useT();
    const handleCopy = () => copy(fieldId);

    return (
        <div className={cn('cr-copyable-text-field', className)}>
            <InputGroup>
                <CustomInput
                    id={fieldId}
                    readOnly
                    inputValue={value}
                    handleClick={handleCopy}
                    type="text"
                    isDisabled={disabled}
                    label={label || ''}
                    defaultLabel={label || ''}
                    placeholder={label || ''}
                />
                <InputGroup.Append>
                    <IconButton onClick={handleCopy} disabled={disabled} title={t('page.body.profile.content.copyLink')}>
                        <CopyIcon />
                    </IconButton>
                </InputGroup.Append>
            </InputGroup>
        </div>
    );
};
