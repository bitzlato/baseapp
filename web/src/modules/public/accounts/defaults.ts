import { createMoney } from 'src/helpers/money';
import { DEFAULT_CURRENCY } from '../currencies/defaults';
import { AccountBalance } from './types';

export const defaultAccount: AccountBalance = {
  ...DEFAULT_CURRENCY,
  currency: DEFAULT_CURRENCY,
  balance: createMoney(0, DEFAULT_CURRENCY),
  locked: createMoney(0, DEFAULT_CURRENCY),
  limit_24_hour: createMoney(0, DEFAULT_CURRENCY),
  limit_1_month: createMoney(0, DEFAULT_CURRENCY),
};
