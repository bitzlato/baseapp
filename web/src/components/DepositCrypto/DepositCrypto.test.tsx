import { shallow } from 'enzyme';
import * as React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { defaultCurrency } from 'src/modules/public/currencies/defaults';
import { defaultWallet } from 'src/modules/user/wallets/defaults';
import { rootReducer } from '../../modules';
import { CopyableTextField } from '../CopyableTextField';
import { DepositCrypto } from './';

const store = createStore(rootReducer);

describe('DepositCrypto', () => {
  let wrapper: any;
  const handleOnCopy = jest.fn();
  const handleGenerateAddress = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <Provider store={store}>
        <DepositCrypto
          dimensions={118}
          error={'error123'}
          handleGenerateAddress={handleGenerateAddress}
          handleOnCopy={handleOnCopy}
          wallet={defaultWallet}
          disabled={false}
          currency={defaultCurrency}
        />
        ,
      </Provider>,
    );
  });

  it('should contains QRCode', () => {
    expect(wrapper.find('.qr-code-wrapper')).toBeTruthy();
  });

  it('should contains CopyableTextField', () => {
    expect(wrapper.find(CopyableTextField)).toBeTruthy();
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
