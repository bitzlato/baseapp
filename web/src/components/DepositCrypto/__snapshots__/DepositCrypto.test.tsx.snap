// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DepositCrypto should match snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <DepositCrypto
    wallet={
      Object {
        "balance": Object {
          "amount": "0.0000",
          "currency": Object {
            "code": "",
            "minorUnit": 4,
          },
        },
        "blockchain_ids": Array [],
        "cc_code": "",
        "code": "",
        "currency": Object {
          "code": "",
          "minorUnit": 4,
        },
        "deposit_enabled": true,
        "deposit_fee": Object {
          "amount": "0.0000",
          "currency": Object {
            "code": "",
            "minorUnit": 4,
          },
        },
        "description": null,
        "homepage": null,
        "icon_id": "",
        "icon_url": "",
        "id": "",
        "limit_1_month": Object {
          "amount": "0.0000",
          "currency": Object {
            "code": "",
            "minorUnit": 4,
          },
        },
        "limit_24_hour": Object {
          "amount": "0.0000",
          "currency": Object {
            "code": "",
            "minorUnit": 4,
          },
        },
        "locked": Object {
          "amount": "0.0000",
          "currency": Object {
            "code": "",
            "minorUnit": 4,
          },
        },
        "min_deposit_amount": Object {
          "amount": "0.0000",
          "currency": Object {
            "code": "",
            "minorUnit": 4,
          },
        },
        "min_withdraw_amount": Object {
          "amount": "0.0000",
          "currency": Object {
            "code": "",
            "minorUnit": 4,
          },
        },
        "minorUnit": 4,
        "name": "",
        "position": 0,
        "precision": 8,
        "price": "1.0",
        "type": "coin",
        "withdraw_fee": Object {
          "amount": "0.0000",
          "currency": Object {
            "code": "",
            "minorUnit": 4,
          },
        },
        "withdraw_limit_24h": "0",
        "withdraw_limit_72h": "0",
        "withdrawal_enabled": true,
      }
    }
  />
  ,
</ContextProvider>
`;
